.github/workflows/automated.yaml

```diff
@@ -0,0 +1,63 @@
+name: Release from React Native
+
+on:
+  repository_dispatches:
+    types: [release]
+
+# NOTE: Changes here need to be reflected in release.yaml
+jobs:
+  publish_template_automation:
+    runs-on: ubuntu-latest
+    env:
+      VERSION: ${{ github.events.client_payload.version }}
+    steps:
+      - name: Safeguard against branch name
+        run: |
+          if [[ "$BRANCH" != *-stable ]]; then
+            echo "Error: This workflow can only be executed from a branch ending with '-stable'."
+            exit 1
+          fi
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+      - name: Move to MAJOR.MINOR-stable or create if it doesn't exist
+        run: |
+          VERSION=$(grep -E '\d+\.\d+' <<< "$VERSION" | awk '{ print "branch="$0"-stable" }')
+          if ! git ls-remote -q --exit-code --heads origin $VERSION > /dev/null; then
+            echo "Creating missing release branch \"$VERSION\""
+            git checkout -B "$VERSION"
+            git push --set-upstream origin "$VERSION"
+          else
+            git checkout "$VERSION"
+          fi
```

Comment:
you don't need to do that. The checkout will happen in the right branch automatically.